sync/src/chain.rs

```diff
@@ -1271,7 +1282,44 @@ impl ChainSync {
 		sent
 	}
 
+	/// propagates new transactions to all peers
+	fn propagate_new_transactions(&mut self, io: &mut SyncIo) -> usize {
+
+		// Early out of nobody to send to.
+		if self.peers.len() == 0 {
+			return 0;
+		}
+
+		let mut packet = RlpStream::new_list(self.transactions_to_send.len());
+		for tx in self.transactions_to_send.iter() {
+			packet.append_raw(tx, 1);
+		}
+		self.transactions_to_send.clear();
+		let rlp = packet.out();
+
+		let lucky_peers = {
+			// sqrt(x)/x scaled to max u32
+			let fraction = (self.peers.len() as f64).powf(-0.5).mul(u32::max_value() as f64).round() as u32;
```

Comment:
at some point we should refactor this peer selection code into a common function